apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -178,7 +178,7 @@ microseconds_to_datetime(MicroSeconds) when is_integer(MicroSeconds) ->
 generate_message_id() ->
     {ok, NodeId} = ejabberd_node_id:node_id(),
     %% Use monotone function here.
-    encode_compact_uuid(now_to_microseconds(now()), NodeId).
+    encode_compact_uuid(now_to_microseconds(p1_time_compat:timestamp()), NodeId).
```

Comment:
We do need unique timestamp here. The whole `ID` is later used as `primary` key in DB.